src/Core/Settings.h

```diff
@@ -896,16 +896,17 @@ class IColumn;
     OBSOLETE_SETTINGS(M)       \
     FORMAT_FACTORY_SETTINGS(M)
 
-DECLARE_SETTINGS_TRAITS_ALLOW_CUSTOM_SETTINGS(SettingsTraits, LIST_OF_SETTINGS)
+#define LIST_OF_ALIASES(M) \
+    M("replication_alter_partitions_sync", "alter_sync") \
+
+DECLARE_SETTINGS_TRAITS_ALLOW_CUSTOM_SETTINGS(SettingsTraits, LIST_OF_SETTINGS, LIST_OF_ALIASES)
```

Comment:
To avoid creating this additional `LIST_OF_ALIASES` and `NO_ALIASES` macro you can instead do something like this:
```cpp
#define COMMON_SETTINGS(M, ALIAS) \
  M(UInt64, alter_sync, 1, "Text", 0) ALIAS(replication_alter_partitions_sync)
  ```
It would require rework of existing code generation logic, but not too hard I guess. Here is possible implementation draft https://godbolt.org/z/c9bYbozaY. WDYT?

Reply:
Great, even more compact! The only thing is that now it should be in "quotes". Not sure what is better. At first we would not have more than single alias per setting. And it is different from original declaration w/o quotes. So, up to you, I like both variants.